### PR TITLE
[Sprint 1] Feature logic main screen

### PR DESCRIPTION
**Pull request by tranminhtri911**

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 1]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+                    default:
+                        break;
+                    //TODO change later
+                }
+                return true;
+            }
+        };
+    }
+
+    @IntDef({
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 2]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+            Item.ITEM_PERSONAL, Item.ITEM_SETUP_PROFILE, Item.ITEM_WORKING_CALENDAR,
+            Item.ITEM_HOLIDAY_CALENDAR, Item.ITEM_STATISTIC_OF_PERSONAL, Item.ITEM_OVERTIME,
+            Item.ITEM_COME_LATE_LEAVE_EARLY, Item.ITEM_WORKSPACE_DATA
+    })
+
+    private @interface Item {
+        int ITEM_PERSONAL = R.id.item_personal;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 3]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_SETUP_PROFILE = R.id.item_setup_profile;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 4]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_WORKING_CALENDAR = R.id.item_working_calendar;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 5]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_HOLIDAY_CALENDAR = R.id.item_holiday_calendar;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 6]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_STATISTIC_OF_PERSONAL = R.id.item_statistic_of_personal;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 7]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_OVERTIME = R.id.item_overtime;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 8]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_COME_LATE_LEAVE_EARLY = R.id.item_come_late_leave_early;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 9]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int ITEM_WORKSPACE_DATA = R.id.item_workspace_data;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java (hunk 1 above)
Comment (tranminhtri911) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: Reformat code rồi nó vẫn báo đó anh

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 10]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+    }
+
+    @IntDef({
+            Item.PERSONAL, Item.SETUP_PROFILE, Item.WORKING_CALENDAR, Item.HOLIDAY_CALENDAR,
+            Item.STATISTIC_OF_PERSONAL, Item.OVERTIME, Item.COME_LATE_LEAVE_EARLY,
+            Item.WORKSPACE_DATA
+    })
+
+    private @interface Item {
+        int PERSONAL = R.id.item_personal;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 11]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int SETUP_PROFILE = R.id.item_setup_profile;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 12]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int WORKING_CALENDAR = R.id.item_working_calendar;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 13]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int HOLIDAY_CALENDAR = R.id.item_holiday_calendar;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 14]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int STATISTIC_OF_PERSONAL = R.id.item_statistic_of_personal;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 15]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int OVERTIME = R.id.item_overtime;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 16]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int COME_LATE_LEAVE_EARLY = R.id.item_come_late_leave_early;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 17]
```diff
@@ -22,4 +32,39 @@ public void onStart() {
...
+        int WORKSPACE_DATA = R.id.item_workspace_data;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java [hunk 18]
```diff
@@ -1,7 +1,12 @@
 package com.framgia.wsm.screen.main;
 
+import android.annotation.SuppressLint;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java]: ERROR: Unused import: android.annotation.SuppressLint. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java [hunk 19]
```diff
@@ -47,4 +55,17 @@ protected void onStop() {
     public MainComponent getMainComponent() {
         return mMainComponent;
     }
+
+    public void initView() {
+        Toolbar toolbar = mBinding.toolbar;
+        mDrawerLayout = mBinding.drawerLayout;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java]: ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field com.framgia.wsm.screen.main.MainActivity.mDrawerLayout from instance method com.framgia.wsm.screen.main.MainActivity.initView()
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD

File: app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java (hunk 18 above)
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.annotation.SuppressLint'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java [hunk 20]
```diff
@@ -15,7 +20,8 @@
 
     @Inject
     MainContract.ViewModel mViewModel;
-
+    private ActivityMainBinding mBinding;
+    private static DrawerLayout mDrawerLayout;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java]: WARNING: [StaticFieldLeak] Do not place Android context classes in static fields; this is a memory leak (and also breaks Instant Run)
 A static field will leak contexts.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 21]
```diff
@@ -22,4 +35,50 @@ public void onStart() {
...
+    public boolean isStatusDrawer() {
+        return mStatusDrawer;
+    }
+
+    private void setStatusDrawer(boolean statusDrawer) {
+        mStatusDrawer = statusDrawer;
+        notifyPropertyChanged(BR.statusDrawer);
+    }
+
+    @IntDef({
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 22]
```diff
@@ -22,4 +37,57 @@ public void onStart() {
...
+
+    public String getUsername() {
+        return mUser != null ? mUser.getName() : "";
+    }
+
+    public String getEmail() {
+        return mUser != null ? mUser.getEmail() : "";
+    }
+
+    @IntDef({
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 23]
```diff
@@ -22,4 +37,57 @@ public void onStart() {
...
+            Item.PERSONAL, Item.SETUP_PROFILE, Item.WORKING_CALENDAR, Item.HOLIDAY_CALENDAR,
+            Item.STATISTIC_OF_PERSONAL, Item.OVERTIME, Item.COME_LATE_LEAVE_EARLY,
+            Item.WORKSPACE_DATA
+    })
+    private @interface Item {
+        int PERSONAL = R.id.item_personal;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 24]
```diff
@@ -22,4 +37,57 @@ public void onStart() {
...
+        int SETUP_PROFILE = R.id.item_setup_profile;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 25]
```diff
@@ -22,4 +37,57 @@ public void onStart() {
...
+        int WORKING_CALENDAR = R.id.item_working_calendar;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/data/model/User.java [hunk 26]
```diff
@@ -0,0 +1,26 @@
+package com.framgia.wsm.data.model;
+
+/**
+ * Created by tri on 24/05/2017.
+ */
+
+public class User {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/model/User.java]: WARNING: [ShortClassName] Avoid short class names like User
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#ShortClassName

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 27]
```diff
@@ -1,16 +1,31 @@
 package com.framgia.wsm.screen.main;
 
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+import android.support.design.widget.NavigationView;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: Unused import: android.support.design.widget.NavigationView. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.support.design.widget.NavigationView'
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 28]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        return mUser != null ? mUser.getEmail() : "";
+    }
+
+    @IntDef({
+            Item.PERSONAL, Item.SETUP_PROFILE, Item.WORKING_CALENDAR, Item.HOLIDAY_CALENDAR,
+            Item.STATISTIC_OF_PERSONAL, Item.OVERTIME, Item.COME_LATE_LEAVE_EARLY,
+            Item.WORKSPACE_DATA
+    })
+    private @interface Item {
+        int PERSONAL = 0;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 29]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int SETUP_PROFILE = 1;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 30]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int WORKING_CALENDAR = 2;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 31]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int HOLIDAY_CALENDAR = 3;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 32]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int STATISTIC_OF_PERSONAL = 4;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 33]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int OVERTIME = 5;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 34]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int COME_LATE_LEAVE_EARLY = 6;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 35]
```diff
@@ -22,4 +39,85 @@ public void onStart() {
...
+        int WORKSPACE_DATA = 7;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 36]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        return mUser != null ? mUser.getEmail() : "";
+    }
+
+    @IntDef({
+            Item.PERSONAL, Item.SETUP_PROFILE, Item.WORKING_CALENDAR, Item.HOLIDAY_CALENDAR,
+            Item.STATISTIC_OF_PERSONAL, Item.OVERTIME, Item.COME_LATE_LEAVE_EARLY,
+            Item.WORKSPACE_DATA
+    })
+    private @interface Item {
+        int PERSONAL = 0;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 37]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int SETUP_PROFILE = 1;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 38]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int WORKING_CALENDAR = 2;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 39]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int HOLIDAY_CALENDAR = 3;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 40]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int STATISTIC_OF_PERSONAL = 4;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 41]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int OVERTIME = 5;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 42]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int COME_LATE_LEAVE_EARLY = 6;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 43]
```diff
@@ -22,4 +36,92 @@ public void onStart() {
...
+        int WORKSPACE_DATA = 7;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.

**Pull request merged**